lisa/wa.py

```diff
@@ -545,3 +547,35 @@ def __init__(self, wa_output, trace_to_df, **kwargs):
     def _get_artifact_df(self, path):
         trace = Trace(path, **self._trace_kwargs)
         return self._trace_to_df(trace)
+
+class WAJankbenchCollector(WAArtifactCollectorBase):
+    """
+    WA collector for the jankbench frame timings.
+
+    Each job will return a single DF which contains one row for each rendered frame.
+    The columns reported are:
+        * total_duration - (float) time in milliseconds to complete the frame
+        * jank_frame     - (int)   1 for a missed deadline, 0 otherwise
+        * name           - (string)subtest name from Jankbench
+        * frame_id       - (int)   frame order, increasing
+
+    An example plotter can be found in the jupyter notebook working directory
+    at <ipynb>/wltests/WAOutput-JankbenchDemo.ipynb
+    """
+
+    NAME = 'jankbench'
+    _ARTIFACT_NAME = "jankbench-results"
+    _ARTIFACT_METRIC_SUFFIX_UNIT = defaultdict(
+        # Default is an empty string
+        str,
+        {
+            'total_duration': 'ms',
+        },
+    )
+
+    def _get_artifact_df(self, path):
```

Comment:
I noticed that this ends up throwing an error when we open the collector and the results set contains non-jankbench workloads. I probably only want an error if the workload should be expected to have jankbench results present and we failed to open then...

What would be the recommended way to deal with that - return None from _get_artifact_df and throw for errors in loading?

Reply:
> I noticed that this ends up throwing an error when we open the collector and the results set contains non-jankbench workloads

In `WAArtifactCollectorBase`:
```
    def _get_job_df(self, job):
        path = job.get_artifact_path(self._ARTIFACT_NAME)
        df = self._get_artifact_df(path)
        return self._add_job_info(job, df)
```
Do you mean that some jobs may not have run Jankbench, but we end up querying `job.get_artifact_path(self._ARTIFACT_NAME)` on it, which returns something unexpected (None?) and makes your method fail ?

Reply:
From what I can see, we already handle any exception raised in `_get_job_df()` in `_get_df()`:
```
            try:
                df = loader(job)
            except Exception as e:
                self.logger.error('Could not load {} dataframe for job {}: {}'.format(self.NAME, job, e))
                return None
```
(`loader()` calls `self._get_job_df(job)`)
If there is a way to tell from a job that it should not be expected to have a given artifact, we could avoid having the logging.error().

Reply:
Checkout this branch and I'll merge it if it solves the problem:
https://github.com/ARM-software/lisa/pull/1538

Reply:
It looks good to me, I'll try it out and let you know

Reply:
Fixed by: https://github.com/ARM-software/lisa/pull/1538